tests/test_request_handling.py

```diff
@@ -511,6 +524,141 @@ def generate_raw_response(
                     f"Wrong finish reason for version: {prompt_template.version} | test: `{test_case['test_aim']}`",
                 )
 
+    async def test_request_handling_streaming(self):
+        async def wrap_generator(tokenizer, raw_response, test_case):
+            tokenized_response = tokenizer.encode(
+                raw_response, add_special_tokens=False
+            )
+            # Append None to the end to produce finish_reason
+            tokenized_response.append(None)
+
+            prev_ids = []
+            for token_id in tokenized_response:
+                if token_id is not None:
+                    offset = tokenizer.decode(prev_ids + [token_id])[
+                        len(tokenizer.decode(prev_ids)) :
+                    ]
+                    prev_ids.append(token_id)
+                    yield offset, None
+                else:
+                    yield "", test_case["expected_finish_reason"]
+
+        for prompt_template in self.test_prompt_templates:
+            tokenizer = AutoTokenizer.from_pretrained(
+                prompt_template.tokenizer_name_or_path
+            )
```

Comment:
# should use this to make sure that added tokens are added
    special_tokens = {"additional_special_tokens": prompt_template.get_additional_tokens()}
    tokenizer.add_special_tokens(special_tokens)